### PR TITLE
change hard-corded part of viz tab behavior

### PR DESCRIPTION
**Pull request by moontrip**
Following @bobular 's opinion, I have changed the hard-corded part, `/visualize/pass-through` to use a programmatic approach using regex.

### CODE REVIEW

**Comment by moontrip** [comment]
@jtlong3rd Is it better to use `useMemo` in this case as well, similar to previous `inFullScreenVisualization` ? Like:

```
  // check whether a user is at visualization
  const inVisualization = useMemo(() => {
    return (
      location.pathname
      .replace(routeBase, '')
      .includes('/visualizations')
    );
  }, [location, routeBase]);
```

**Comment by moontrip** [comment]
@bobular After contemplating your comment at other PR ([here](https://github.com/VEuPathDB/web-eda/pull/192#issuecomment-871435562)), I think I got your point what it meant. Please correct me if I am wrong.

So, basically my latest change will work for the case, `/viz/pass-through/UUID`, but not work for , e.g., `/viz/computation/UUID` (not currently existing but may be added in the future). Is it what you thought?

**Comment by bobular** [comment]
Sorry @moontrip - you are right - the simple version seems to work just fine.  For some reason (without testing it) I thought it would not.

**Comment by bobular** [comment]
> Sorry @moontrip - you are right - the simple version seems to work just fine. For some reason (without testing it) I thought it would not.

Though is the `   .replace(routeBase, '') ` needed?

**Comment by moontrip** [comment]
@bobular it is not a must as I doubt we will somehow use the string `visualizations` twice in a URL

**Comment by bobular** [comment]
> @bobular it is not a must as I doubt we will somehow use the string `visualizations` twice in a URL

Aha - yes, good thinking

**Comment by jtlong3rd** [comment]
> @jtlong3rd Is it better to use `useMemo` in this case as well, similar to previous `inFullScreenVisualization` ? Like:
> 
> ```
>   // check whether a user is at visualization
>   const inVisualization = useMemo(() => {
>     return (
>       location.pathname
>       .replace(routeBase, '')
>       .includes('/visualizations')
>     );
>   }, [location, routeBase]);
> ```

I think what you have is fine! `useMemo` is helpful when you want to (1) (2) preservce the 

> @jtlong3rd Is it better to use `useMemo` in this case as well, similar to previous `inFullScreenVisualization` ? Like:
> 
> ```
>   // check whether a user is at visualization
>   const inVisualization = useMemo(() => {
>     return (
>       location.pathname
>       .replace(routeBase, '')
>       .includes('/visualizations')
>     );
>   }, [location, routeBase]);
> ```

I actually like what you have right now! I realized yesterday that the `useMemo` which I suggested yesterday was overkill - I'll leave a comment in that PR when I have a moment.

**Review by jtlong3rd — APPROVED** [review]

File: src/lib/workspace/AnalysisPanel.tsx [hunk 1]
```diff
@@ -120,9 +108,11 @@ export function AnalysisPanel(props: Props) {
           },
           {
             display: 'Visualize',
-            // check whether user is at viz's full screen mode
-            route: inFullscreenVisualization
-              ? '/visualizations/pass-through'
+            // check whether user is at viz
+            route: location.pathname
+              .replace(routeBase, '')
+              .includes('/visualizations')
```
Comment (dmfalke) [inline src/lib/workspace/AnalysisPanel.tsx:114]: I think we should keep the replace, but the following can probably be replaced with a `startsWith`:

```suggestion
              .replace(routeBase, '')
              .startsWith('/visualizations')
```

**Review by dmfalke — CHANGES_REQUESTED** [review]
Part of me wants to update the props api for `WorkspaceNavigation` such that you can use a function for the value of `route`, giving the consumer access to whether the tab is active or not.

E.g., 
```typescript
interface WorkspaceNavigationProps {
  display: 'Visualize',
  route: ({ isActive }) => '/visualizations' + (isActive ? lastVizPath : '')
}
```

But not sure it's worth it :slightly_smiling_face:.

I left a suggestion inline.

**Comment by moontrip** [comment]
@jtlong3rd Thank you for your comment and suggestion 👍

File: src/lib/workspace/AnalysisPanel.tsx (hunk 1 above)
Comment (moontrip) [reply src/lib/workspace/AnalysisPanel.tsx:114]: @dmfalke Thank you for your suggestion to change and I agree with that to use startsWith() 👍 May I ask your opinion on one more thing ? 😊 I am just debating myself, whether I will stick to the current approach you suggested or another approach to remove the last url element (e.g., using regEx). The former can certainly do what we want now, but the latter can handle more general cases as described follows.

For example, current approach will work for the current case, /viz/pass-through/UUID, but not work for more general case, e.g., /viz/computation/UUID (not currently existing but may probably be added in the future). The second approach (removing the last url element) would be able to handle general cases like `location.pathname.replace(routeBase, '').replace(/\/[^/]*$/, '')`

Certainly we can revisit this by sticking to the current one, though. What do you think?

Comment (dmfalke) [reply src/lib/workspace/AnalysisPanel.tsx:114]: I think this currently approach is actually the behavior we will want. In cases where there are multiple computations offered, clicking on the tab should go to the computation selector.

Comment (moontrip) [reply src/lib/workspace/AnalysisPanel.tsx]: @dmfalke thank you for your info 👍 I have updated with using startsWith() instead of includes()

**Review by bobular — APPROVED** [review]
Looks good - approved!

**Review by dmfalke — APPROVED** [review]

**Pull request merged**